spec/supervisor_spec.rb

```diff
@@ -161,4 +161,117 @@ def connect task, core_versions:, sxl_version:
+        }
+        
+        # Test the logic that would be used in accept_connection
+        if site_settings && site_settings['type'] == 'tlc'
+          proxy = RSMP::TLC::TrafficControllerProxy.new settings.merge(site_id: 'TLC001')
+        else
+          proxy = RSMP::SiteProxy.new settings.merge(site_id: 'TLC001')
+        end
+        
+        expect(proxy).to be_an(RSMP::TLC::TrafficControllerProxy)
+      end
+    end
+
+    context 'when site uses guest configuration with tlc' do
+      it 'creates TLCProxy when guest sxl is tlc' do
+        # Test the core proxy creation logic directly
+        # Create a basic supervisor for the test
+        test_supervisor = RSMP::Supervisor.new(
+          supervisor_settings: { 'port' => 13115, 'guest' => { 'sxl' => 'tlc', 'type' => 'tlc' } },
+          log_settings: log_settings
+        )
+        
+        # Mock settings that would be passed to the proxy constructor
+        settings = {
+          supervisor: test_supervisor,
+          ip: '127.0.0.1',
+          port: 12345,
+          socket: double('socket'),
+          stream: double('stream'),
+          protocol: double('protocol'),
+          logger: double('logger'),
+          archive: double('archive')
+        }
+        
+        # Test with TLC-like settings
+        tlc_site_settings = { 'sxl' => 'tlc', 'type' => 'tlc' }
+        non_tlc_site_settings = { 'sxl' => 'core', 'type' => 'core' }
+        
+        # Test TLC proxy creation
+        tlc_proxy = RSMP::TLC::TrafficControllerProxy.new settings.merge(site_id: 'TLC001')
+        
+        # Test non-TLC proxy creation
+        non_tlc_proxy = RSMP::SiteProxy.new settings.merge(site_id: 'OTHER001')
+        
+        expect(tlc_proxy).to be_an(RSMP::TLC::TrafficControllerProxy)
+        expect(non_tlc_proxy).to be_an(RSMP::SiteProxy)
+        expect(non_tlc_proxy).not_to be_an(RSMP::TLC::TrafficControllerProxy)
+      end
+    end
+
+    context 'when guest configuration is not tlc' do
+      it 'creates regular SiteProxy for non-TLC guest sites' do
+        # Use basic supervisor with non-TLC guest settings
+        supervisor = RSMP::Supervisor.new(
+          supervisor_settings: supervisor_settings,  # This has guest sxl as 'tlc', which will get non-TLC treatment in this test
+          log_settings: log_settings
+        )
+        
+        # Manually override the site settings to simulate non-TLC
+        site_settings = { 'sxl' => 'other_type' }  # This won't be validated since we're testing logic directly
+        
+        settings = {
+          supervisor: supervisor,
+          ip: '127.0.0.1',
+          port: 12345,
+          socket: double('socket'),
+          stream: double('stream'),
+          protocol: double('protocol'),
+          logger: double('logger'),
+          archive: double('archive')
+        }
+        
+        # Test the logic for non-TLC sites
+        if site_settings && site_settings['type'] == 'tlc'
```

Comment:
conditionals based on a settings we just set up makes no sense. please reconsider these tests